### ISSUE TITLE
Coordinates of caption elements

### ISSUE THREAD

**Issue by keto33** [kermitt2/grobid#1008]
This may seem unnecessary, but it should be a feasible feature suggestion.

GROBID outputs all coordinates of structures except for text blocks. I am mostly interested in the coordinates of figure captions. When figures are embedded as EPS in vector format rather than raster/bitmap, GROBID does not correctly detect the bounding box of the figure, as drawings and texts are somehow blended into the PDF structure rather than being a distinguishable stream. In such cases, the bounding box of the figure caption can be helpful in estimating the actual bounding box of the EPS figure.

**Comment by kermitt2** [comment]
Hi @keto33 !

Thanks for the issue. 

> GROBID outputs all coordinates of structures except for text blocks. 

Yes text blocks are not part of the TEI XML output because they are presentation/layout elements, not something related to the logicial structure of the document (like paragraphs, titles, etc.). 

> I am mostly interested in the coordinates of figure captions. When figures are embedded as EPS in vector format rather than raster/bitmap, GROBID does not correctly detect the bounding box of the figure, as drawings and texts are somehow blended into the PDF structure rather than being a distinguishable stream.

Yes the coordinates of the caption elements are indeed not outputted  currently and there is no reason not to do it. 

Regarding the "graphic part" of a figure, this is more or less implemented in PR #963 (the whole PR is not usable at this stage, really work in progress), the vector graphics are further analyzed to detect their boundaries, deal with overlapped text, etc. so that we have reliable "figure graphic" aggregated elements similar to the embedded bitmaps. There are many other things in this PR and it will take a lot time to be completed !

**Comment by ClementFrvl** [comment]
Hello!

Is there an ongoing effort or a specific branch where coordinates of text blocks can be extracted as part of the TEI/XML output?

I checked the documentation and I saw p elements are under teiCoordinates, and I am running this command:

```bash
curl --form input=@./Papers/test.pdf --form teiCoordinates='head' --form teiCoordinates='p' host:8070/api/processFulltextDocument
```

However there are no coordinates for the <p>, which I'm interested in.
<img width="507" alt="image" src="https://github.com/user-attachments/assets/738396b8-c8e0-4630-aabe-76588a0e875e">

Please let me know if there is a solution or anything I can do to assist!

**Comment by lfoppiano** [comment]
Hi @ClementFrvl, which version are you using? This seems a problem of grobid version 0.8.0 which disappears on the grobid master's version. 🤔

**Comment by ClementFrvl** [comment]
Hey, I am using 0.8.0, that may be the reason why. 

My server is ARM-based though, I just tried with version 0.7.3, but I'm having the same issue.

<img width="478" alt="image" src="https://github.com/user-attachments/assets/457faeb1-7cfd-4639-b754-597ead8a05ce">

Is there a newer arm version available ?

**Comment by lfoppiano** [comment]
We're working on a new version since a few weeks, hopefully we will be able to release soon.

**Comment by lfoppiano** [comment]
Should be solved in version 0.8.1
